framework/src/transfers/MultiAppInterpolationTransfer.C

```diff
@@ -66,492 +83,498 @@ MultiAppInterpolationTransfer::MultiAppInterpolationTransfer(const InputParamete
 }
 
 void
-MultiAppInterpolationTransfer::execute()
+MultiAppInterpolationTransfer::subdomainIDsNode(MooseMesh & mesh,
+                                                Node & node,
+                                                std::set<subdomain_id_type> & subdomainids)
 {
-  _console << "Beginning InterpolationTransfer " << name() << std::endl;
-
-  switch (_current_direction)
-  {
-    case TO_MULTIAPP:
-    {
-      FEProblemBase & from_problem = _multi_app->problemBase();
-      MooseVariableFEBase & from_var = from_problem.getVariable(
-          0, _from_var_name, Moose::VarKindType::VAR_ANY, Moose::VarFieldType::VAR_FIELD_STANDARD);
+  // We need this map to figure out to which subdomains a given mesh point is attached
+  auto & node_to_elem = mesh.nodeToElemMap();
+  auto node_to_elem_pair = node_to_elem.find(node.id());
 
-      MeshBase * from_mesh = NULL;
+  if (node_to_elem_pair == node_to_elem.end())
+    mooseError("Can not find elements for node ", node.id());
 
-      if (_displaced_source_mesh && from_problem.getDisplacedProblem())
-        from_mesh = &from_problem.getDisplacedProblem()->mesh().getMesh();
-      else
-        from_mesh = &from_problem.mesh().getMesh();
+  subdomainids.clear();
+  // Add all subdomain IDs that are attached to node
+  for (auto element : node_to_elem_pair->second)
+  {
+    auto & elem = mesh.getMesh().elem_ref(element);
+    auto subdomain = elem.subdomain_id();
 
-      SystemBase & from_system_base = from_var.sys();
-      System & from_sys = from_system_base.system();
+    subdomainids.insert(subdomain);
+  }
+}
 
-      unsigned int from_sys_num = from_sys.number();
-      unsigned int from_var_num = from_sys.variable_number(from_var.name());
+void
+MultiAppInterpolationTransfer::fillSourceInterpolationPoints(
+    FEProblemBase & from_problem,
+    MooseVariableFEBase & from_var,
```

Comment:
Can this be `const MooseVariableFEBase &` ?

Reply:
Also `MooseVariableFEBase` is just now a `typedef` for `MooseVariableFieldBase` with the addition of finite volume variables. Let's use `MooseVariableFieldBase` instead to indicate that this code should be generally applicable to all field variables (both FE and FV)

Reply:
```
 error: 'this' argument to member function 'sys' has type 'const MooseVariableFieldBase', but function is not marked const
  const SystemBase & from_system_base = from_var.sys();
```

I guess I can not make this `const`

Reply:
Get this done by adding:

```
  /**
   * Get the system this variable is part of.
   */
  const SystemBase & sys() const { return _sys; }
```

Reply:
See `const` is a "beautiful" virus that slowly spreads through the code. Because I asked for `const` in your original PR, we ended up labeling `sys()` `const` when it should have been labeled `const` long ago.